app/src/lib/shells/win32.ts

```diff
@@ -215,6 +235,146 @@ async function findGitBash(): Promise<string | null> {
   return null
 }
 
+async function findWslBashShellsCommandLine(): Promise<ReadonlyArray<
+  IFoundShell<Shell>
+> | null> {
+  const promise = new Promise<IFoundShell<Shell>[] | null>(resolve => {
+    //check we have a valid distro installed (ubuntu, opensuse, sles, kali, debian)
+    const windowsRoot = process.env.SystemRoot || 'C:\\Windows'
+    const windowsSystem32 = windowsRoot + '\\System32\\'
+    const defaultWslExe = windowsSystem32 + 'wsl.exe'
+
+    const shells = [
+      {
+        shell: Shell.WslBash,
+        path: defaultWslExe,
+        name: 'WSL Bash (Default)',
+      },
+    ]
+    const wslConfigExe = windowsSystem32 + 'wslconfig.exe'
+
+    const ls = spawn(wslConfigExe, ['/list'])
+    ls.stdout.on('data', data => {
+      //wslconfig.exe /list returns a unicode array - santise to human readable format
+      const trimmedData = data
+        .toString()
+        .replace(
+          /[^A-Za-z 0-9\.,\?""!@#\$%\^&\*\(\)-_=\+;:<>\/\\\|\}\{\[\]`~]*/g,
+          ''
+        )
+
+      //user might have installed WLS but have no valid distro's installed
+      if (trimmedData.search('no installed') === -1) {
+        const wslShellsNames = trimmedData.split(':')
+        if (wslShellsNames.length > 0) {
+          //ignore first line
+          for (let i = 1; i < wslShellsNames.length; i++) {
+            const wslShellNames = wslShellsNames[i].trim()
+
+            //default path for distro exe's
+            const wslShellPathWindows =
+              process.env.HOME + '\\AppData\\Local\\Microsoft\\WindowsApps\\'
+
+            if (wslShellNames.search('Debian') !== -1) {
+              shells.push({
+                shell: Shell.WslDebian,
+                name: 'debian',
+                path: wslShellPathWindows + 'debian.exe',
```

Comment:
Don't use these Distro-specific EXEs, you can use `wsl.exe -d` instead - people can install their own distros via https://github.com/yuk7/wsldl et al, and this way you don't have to get in the game of hard-coding a bunch of distro names

Reply:
Ok i will have a play around with this option